core/src/main/java/com/linecorp/armeria/internal/common/stream/SurroundingPublisher.java

```diff
@@ -67,6 +75,22 @@ public SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher,
             this.publisher = new PublisherBasedStreamMessage<>(publisher);
         }
         this.tail = tail;
+        finalizer = null;
+    }
+
+    @SuppressWarnings("unchecked")
+    private SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher,
```

Comment:
nit: If you want, you can define another constructor that will dedup the assigning code.

Reply:
![image](https://github.com/line/armeria/assets/34854527/0511b72a-1e0a-4917-a925-df44406cfc5e)

```java
public SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher,
                            Function<@Nullable Throwable, ? extends T> finalizer) { .. }

new SurroundingPublisher<>(null, Mono.just(1), null /*tail or finalizer?*/),
```
oh If we add above constructor, my IDE can't select the proper constructor on above `tail: null` case and test failed 😅

Is there any other good way?

Reply:
Doesn't this work? 😄 
```java
public SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher, @Nullable T tail) {
    this(head, publisher, tail, null);
}

private SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher,
                             Function<@Nullable Throwable, ? extends T> finalizer) {
    this(head, publisher, null, requireNonNull(finalizer, "finalizer"));
}

private SurroundingPublisher(
        @Nullable T head, Publisher<? extends T> publisher,
        @Nullable T tail, @Nullable Function<@Nullable Throwable, ? extends T> finalizer) {
    requireNonNull(publisher, "publisher");
    this.head = head;
    if (publisher instanceof StreamMessage) {
        //noinspection unchecked
        this.publisher = (StreamMessage<T>) publisher;
    } else {
        this.publisher = new PublisherBasedStreamMessage<>(publisher);
    }
    this.tail = tail;
    this.finalizer = finalizer;
}
```

Reply:
oh~ this looks better. thanks! ✅ updated